### PR TITLE
copy load infos from existing one

### PR DESCRIPTION
**Pull request by AbdelHedhili**
Signed-off-by: Abdelsalem <abdelsalem.hedhili@rte-france.com>

### CODE REVIEW

File: src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java [hunk 1]
```diff
@@ -82,29 +82,31 @@ public void searchEquipmentInfosMultiVariants() {
...
         EquipmentInfos line2Infos = EquipmentInfos.builder().networkUuid(NETWORK_UUID).id("id_l2").name("name_l2").variantId(VariantManagerConstants.INITIAL_VARIANT_ID).type("LINE").voltageLevels(Set.of(VoltageLevelInfos.builder().id("vl3").name("vl3").build())).build();
         EquipmentInfos twInfos = EquipmentInfos.builder().networkUuid(NETWORK_UUID).id("id_tw1").name("name_tw1").variantId(VariantManagerConstants.INITIAL_VARIANT_ID).type("TWO_WINDINGS_TRANSFORMER").voltageLevels(Set.of(VoltageLevelInfos.builder().id("vl5").name("vl5").build())).build();
+        EquipmentInfos load1Infos = EquipmentInfos.builder().networkUuid(NETWORK_UUID).id("loadId1").name("name_load1").variantId(VariantManagerConstants.INITIAL_VARIANT_ID).type("LOAD").voltageLevels(Set.of(VoltageLevelInfos.builder().id("vl1").name("vl1").build())).build();
+        EquipmentInfos load2Infos = EquipmentInfos.builder().networkUuid(NETWORK_UUID).id("loadId2").name("name_load1").variantId(VariantManagerConstants.INITIAL_VARIANT_ID).type("LOAD").voltageLevels(Set.of(VoltageLevelInfos.builder().id("vl1").name("vl1").build())).build();
```
Comment (sBouzols) [inline src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java:86]: .name("name_load**2**"

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 2]
```diff
@@ -348,6 +348,17 @@ public void initBinder(WebDataBinder webdataBinder) {
         return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(studyService.getLoadsMapData(studyUuid, nodeUuid, substationsIds));
     }
 
+    @GetMapping(value = "/studies/{studyUuid}/nodes/{nodeUuid}/network-map/loads/{loadId}")
+    @Operation(summary = "Get specific load description")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The load data")})
+    public ResponseEntity<Mono<String>> getLoadMapData(
+            @PathVariable("studyUuid") UUID studyUuid,
+            @PathVariable("nodeUuid") UUID nodeUuid,
+            @PathVariable("loadId") String loadUuid) {
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/study/server/StudyController.java:357]: `loadUuid` ->  `loadId`

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 3]
```diff
@@ -348,6 +348,17 @@ public void initBinder(WebDataBinder webdataBinder) {
...
+
+        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(studyService.getLoadMapData(studyUuid, nodeUuid, loadUuid));
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/study/server/StudyController.java:359]: `loadUuid` -> `loadId`

File: src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java (hunk 1 above)
Comment (AbdelHedhili) [reply src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java:86]: fixed

File: src/main/java/org/gridsuite/study/server/StudyController.java (hunk 2 above)
Comment (AbdelHedhili) [reply src/main/java/org/gridsuite/study/server/StudyController.java:357]: fixed

File: src/main/java/org/gridsuite/study/server/StudyController.java (hunk 3 above)
Comment (AbdelHedhili) [reply src/main/java/org/gridsuite/study/server/StudyController.java:359]: fixed

File: src/test/java/org/gridsuite/study/server/StudyTest.java [hunk 4]
```diff
@@ -1425,6 +1433,23 @@ public void testNetworkModificationSwitch() {
...
+        UUID studyNameUserIdUuid = createStudy("userId", CASE_UUID);
+        UUID rootNodeUuid = getRootNodeUuid(studyNameUserIdUuid);
+
+        //get the load map data info of a network
+        webTestClient.get()
+                .uri("/v1/studies/{studyUuid}/nodes/{nodeUuid}/network-map/loads/{loadId}", studyNameUserIdUuid, rootNodeUuid, LOAD_ID_1)
+                .exchange()
+                .expectStatus().isOk()
+                .expectHeader().contentType(MediaType.APPLICATION_JSON)
+                .expectBody(String.class);
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/study/server/StudyTest.java:1448]: check body content ?

File: src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java [hunk 5]
```diff
@@ -153,18 +155,37 @@ public void searchEquipmentInfosMultiVariants() {
...
         hits.clear();
-        studyService.searchEquipments(STUDY_UUID, VARIANT_NODE_UUID, "id_tw1", EquipmentInfosService.FieldSelector.ID).subscribe(hits::add);
+        studyService.searchEquipments(STUDY_UUID, VARIANT_NODE_UUID, "id_tw1", EquipmentInfosService.FieldSelector.ID, null).subscribe(hits::add);
         assertEquals(0, hits.size());
 
         // Search all equipments with node of new variant
         hits.clear();
-        studyService.searchEquipments(STUDY_UUID, VARIANT_NODE_UUID, "id_", EquipmentInfosService.FieldSelector.ID).subscribe(hits::add);
+        studyService.searchEquipments(STUDY_UUID, VARIANT_NODE_UUID, "id_", EquipmentInfosService.FieldSelector.ID, null).subscribe(hits::add);
+        studyService.searchEquipments(STUDY_UUID, VARIANT_NODE_UUID, "id_", EquipmentInfosService.FieldSelector.ID, null).subscribe(hits::add);
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java:188]: why duplicate this line here ?

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/test/java/org/gridsuite/study/server/StudyTest.java (hunk 4 above)
Comment (AbdelHedhili) [reply src/test/java/org/gridsuite/study/server/StudyTest.java:1448]: There is no point, I'm already checking that I called the right url and that URL is returning a mocked answer that isn't even touched by the study-server so the result doesn't matter at all as long as it's present.

Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/study/server/StudyTest.java:1448]: ok

File: src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java (hunk 5 above)
Comment (AbdelHedhili) [reply src/test/java/org/gridsuite/study/server/StudyServiceSearchTests.java:188]: removed

**Review by FranckLecuyer — APPROVED** [review]

**Review by sBouzols — APPROVED** [review]
Code Review OK
Tests OK

**Pull request merged**